deriving-compat.cabal

```diff
@@ -31,6 +32,18 @@ description:         Provides Template Haskell functions that mimic deriving
                      .
                      See the "Data.Deriving" module for a full list of backported changes.
                      .
+                     In addition, @deriving-compat@ also provides functionality
+                     to derive classes in the @Data.Functor.Classes@ module,
+                     covering the @Eq1@, @Eq2@, @Ord1@, @Ord2@, @Read1@,
+                     @Read2@, @Show1@, and @Show2@ classes. This is
+                     functionality that upstream GHC does not currently have,
+                     so strictly speaking, this functionality is outside of the
+                     main scope of @deriving-compat@. Nevertheless, the
+                     underlying Template Haskell functionality needed to derive
+                     @Eq@ and friends extends very naturally to @Eq1@ and
+                     friends, so this extra functionality is included in
+                     @deriving-compat@ as a convenience.
+                     .
```

Comment:
Proposing a rephrasing:
```haskell
{-
@deriving-compat@ module also provides some additional deriving functionality. Because of the work done that is not yet merged into GHC - `Data.Functor.Classes` naturally extends @Eq@, @Ord@, @Read@ @Show@ derivation with derivation of type class instances for @Eq1@, @Eq2@, @Ord1@, @Ord2@, @Read1@, @Read2@, @Show1@, and @Show2@ type classes, through `Template Haskell`. This functionality is outside of the main agenda of the project.
-}
```

Tried to duplicate while preserving all meaning.

If not - at least the "needed to derive @Eq@ and friends extends very naturally to @Eq1@ and friends" - needs to be deduplicated with the real enumeration and assurance of what works by "extending naturally" but not regarded as a main guarantee/agenda of a package.

Reply:
My apologies, I'm not quite sure I understand the concern you're trying to address. Is your concern that because these classes aren't backports of GHC functionality, readers might assume that it will be removed at some point, and you want the documentation to clarify that `deriving-compat` _won't_ remove this functionality?

Reply:
Thank you for being reserved. Satisfied with everything in any case. The main thing that just thought it would be useful to mention unique functionality and I and others would know more about it.

With the rephrasing, just tried to give work to straightforward deliverance.

Thank you.

Reply:
> The main thing that just thought it would be useful to mention unique functionality and I and others would know more about it.

OK, so you're trying to emphasize the fact that this is additional functionality that isn't present in GHC? What about this, then?

> In addition, `deriving-compat` also provides some additional functionality that is not yet available in GHC. `deriving-compat` provides the ability to derive classes in the `Data.Functor.Classes` module, covering the `Eq1`, `Eq2`, `Ord1`, `Ord2`, `Read1`, `Read2`, `Show1`, and `Show2` classes. This functionality is outside of the main scope of `deriving-compat`, but since the underlying Template Haskell functionality needed to derive `Eq` and friends extends very naturally to `Eq1` and friends, this extra functionality is included in `deriving-compat` as a convenience.

(To be clear, I'm not opposed to your suggestion in https://github.com/haskell-compat/deriving-compat/pull/36#discussion_r740097658. I'm just trying to phrase it in a way that sounds natural to a native English speaker.)

Reply:
It also can be:

{-
Beside the main agenda of the project, `Data.Functor.Classes` able to naturally extend GHC deriving into @Eq1@, @Eq2@, @Ord1@, @Ord2@, @Read1@, @Read2@, @Show1@, and @Show2@. It is not yet merged upstream.
-}

`It is not yet merged upstream` - explains generally enough & prevents from logical reports "merge this upstream" "why" & frequent questions about it.

Mentioning `naturally extend GHC deriving GHC into` - talks about GHC-like guarantees, and `beside the main agenda` & `It is not yet merged upstream.` explains the situation more, so people can evaluate the situation.

In any case - it is Ok - details, a material to work with.

Thank you. I wouldn't say more.

Reply:
Thank you for the feedback. I've pushed a change which chooses this wording:

> In addition, `deriving-compat` also provides some additional `deriving` functionality that has not yet been merged into upstream GHC. Aside from the GHC `deriving` extensions mentioned above, `deriving-compat` also permits deriving instances of classes in the `Data.Functor.Classes` module, covering the `Eq1`, `Eq2`, `Ord1`, `Ord2`, `Read1`, `Read2`, `Show1`, and `Show2` classes. This extra functionality is outside of the main scope of `deriving-compat`, as it does not backport extensions that exist in today's GHC. Nevertheless, the underlying Template Haskell machinery needed to derive `Eq` and friends extends very naturally to `Eq1` and friends, so this extra functionality is included in `deriving-compat` as a convenience.

I took some liberties with the exact wording you suggested, but I believe it covers all of the points you raised in https://github.com/haskell-compat/deriving-compat/pull/36#discussion_r740373326. It uses the phrase "Aside from the GHC `deriving` extensions mentioned above", it emphasizes that the `Data.Functor.Classes`–related `deriving` functionality is outside the main scope of `deriving-compat`, and it mentions that this functionality has been yet been merged into upstream GHC.

Reply:
Thank you for this.